livy/src/main/java/org/apache/zeppelin/livy/BaseLivyInterprereter.java

```diff
@@ -219,7 +234,23 @@ public InterpreterResult interpret(String code, boolean displayAppInfo)
       }
       stmtInfo = getStatementInfo(stmtInfo.id);
     }
-    return getResultFromStatementInfo(stmtInfo, displayAppInfo);
+    return appendSessionExpire(getResultFromStatementInfo(stmtInfo, displayAppInfo));
+
+  }
+
+  private InterpreterResult appendSessionExpire(InterpreterResult result) {
+    if (sessionExpired) {
+      InterpreterResult result2 = new InterpreterResult(result.code());
+      for (InterpreterResultMessage message : result.message()) {
+        result2.add(message.getType(), message.getData());
+      }
+      result2.add(InterpreterResult.Type.HTML,
+          "<hr/><font color=\"red\">Previous session is expired, new session is created.</font>");
```

Comment:
perhaps it is better for this message to be at the top of the result output, rather than at the bottom of it? doesn't it require the user to scroll down if he has a long output?